typed/merge.go

```diff
@@ -140,81 +144,89 @@ func (w *mergingWalker) prepareDescent(pe fieldpath.PathElement, tr schema.TypeR
 func (w *mergingWalker) finishDescent(w2 *mergingWalker) {
 	// if the descent caused a realloc, ensure that we reuse the buffer
 	// for the next sibling.
-	w.errorFormatter = w2.errorFormatter.parent()
+	w.path = w2.path[:len(w2.path)-1]
 	*w.spareWalkers = append(*w.spareWalkers, w2)
 }
 
-func (w *mergingWalker) derefMap(prefix string, v *value.Value, dest **value.Map) (errs ValidationErrors) {
+func (w *mergingWalker) derefMap(prefix string, v value.Value, dest *value.Map) (errs ValidationErrors) {
 	// taking dest as input so that it can be called as a one-liner with
 	// append.
 	if v == nil {
 		return nil
 	}
-	m, err := mapValue(*v)
+	m, err := mapValue(v)
 	if err != nil {
-		return w.prefixError(prefix, err)
+		return errorf("%v: %v", prefix, err)
```

Comment:
This substantially changes the way the path is reported in the error, no? Is this way of reporting the path consistent with what WithPrefix does?

Reply:
Not sure which of these you're talking about:
1. We don't collect the path in the same way, agreed.
2. Should I change back to `errorf(prefix + err)`, which is more similar to what we had before?

Reply:
I think we need a mechanism that doesn't rely on every call site doing the right thing with the prefix. I don't care if you change the format. I care a little bit that it's consistent, but that should be the case if you satisfy my first constraint anyway :)